NBitcoin/Protocol/Node.cs

```diff
@@ -1052,6 +1055,9 @@ public void VersionHandshake(NodeRequirement requirements, CancellationToken can
 					return;
 				}
 
+				// Signal ADDRv2 support (BIP155).
+				SendMessageAsync(new SendAddrV2Payload());

```

Comment:
@lontivero The sequence of events in this `VersionHandshake` is as follows:

```
my node ----ver----------> remote node: SendMessageAsync(MyVersion);
my node <---ver----------- remote node: var payload = listener.ReceivePayload<Payload>(cancellationToken); var version = (VersionPayload)payload;
my node ----sendaddrv2---> remote node: SendMessageAsync(new SendAddrV2Payload());
my node ----verack-------> remote node: SendMessageAsync(new VerAckPayload());
my node <---verack-------- remote node: listener.ReceivePayload<VerAckPayload>(cancellationToken);
```

As noted here: https://github.com/MetacoSA/NBitcoin/pull/949#issuecomment-743728913 this isn't what you claim to do (in your first commit message, nor in your second commit message,) but this is correct.

Reply:
The BIP was modified after my commit. However, given I forced the commits it seems like my commit was done one day after the BIP was modified. 
Anyway, the code that enforce the disconnection when the order of the messages is incorrect was not released yet.

Reply:
ACK